native/cocos/renderer/gfx-vulkan/VKCommandBuffer.cpp

```diff
@@ -965,8 +965,8 @@ void CCVKCommandBuffer::pipelineBarrier(const GeneralBarrier *barrier, const Buf
         }
 
         if (!fullBufferBarriers.empty() || !fullImageBarriers.empty()) {
-            vkCmdPipelineBarrier(_gpuCommandBuffer->vkCommandBuffer, fullSrcStageMask, fullDstStageMask, 0, 0, pMemoryBarrier,
-                                 fullBufferBarriers.size(), fullBufferBarriers.data(), fullImageBarriers.size(), fullImageBarriers.data());
+            vkCmdPipelineBarrier(_gpuCommandBuffer->vkCommandBuffer, fullSrcStageMask, fullDstStageMask, 0, pMemoryBarrier ? 1 : 0,
+                                 pMemoryBarrier, fullBufferBarriers.size(), fullBufferBarriers.data(), fullImageBarriers.size(), fullImageBarriers.data());
```

Comment:
Shall we change the `if (!fullBufferBarriers.empty() || !fullImageBarriers.empty())` to `if (!fullBufferBarriers.empty() || !fullImageBarriers.empty() || pMemoryBarrier)` ?

`pMemoryBarrier` might not be executed.

Reply:
personally i think it's fine, because `memoryBarrierCount` arg for `vkCmdPipelineBarrier` decides it will be used or not.